test/report-design-viewer/index.js

```diff
@@ -0,0 +1,120 @@
+var EventEmitter = require('events').EventEmitter;
+var ttyDecorator = require('../../lib/reporters/errors/decorators/tty');
+var reporters    = require('../../lib/reporters/index');
+var util         = require('util');
+var errs         = require('./errors');
+
+var testMocks = [
+    {
+        name:    'fixture1Test1',
+        fixture: {
+            name: 'fixture1',
+            path: './fixture1.js'
+        }
+    },
+    {
+        name:    'fixture1Test2',
+        fixture: {
+            name: 'fixture1',
+            path: './fixture1.js'
+        }
+    },
+    {
+        name:    'fixture2Test1',
+        fixture: {
+            name: 'fixture2',
+            path: './fixture2.js'
+        }
+    },
+    {
+        name:    'fixture2Test2',
+        fixture: {
+            name: 'fixture2',
+            path: './fixture2.js'
+        }
+    },
+    {
+        name:    'fixture3Test1',
+        fixture: {
+            name: 'fixture3',
+            path: './fixture3.js'
+        }
+    }
+];
+
+var browserConnectionMock = { userAgent: 'Chrome 15.0.874 / Mac OS X 10.8.1' };
+
+// Task mock
+var TaskMock = function () {
+    EventEmitter.call(this);
+
+    this.tests              = testMocks;
+    this.browserConnections = [browserConnectionMock];
+};
+
+util.inherits(TaskMock, EventEmitter);
+
+var testRunMocks = [
+    {
+        test:              testMocks[0],
+        unstable:          false,
+        browserConnection: browserConnectionMock,
+        errs:              errs[0]
+    },
+    {
+        test:              testMocks[1],
+        unstable:          false,
+        browserConnection: browserConnectionMock,
+        errs:              errs[1]
+    },
+    {
+        test:              testMocks[2],
+        unstable:          true,
+        browserConnection: browserConnectionMock,
+        errs:              errs[2]
+    },
+    {
+        test:              testMocks[3],
+        unstable:          false,
+        browserConnection: browserConnectionMock,
+        errs:              []
+    },
+    {
+        test:              testMocks[4],
+        unstable:          false,
+        browserConnection: browserConnectionMock,
+        errs:              []
+    }
+];
+
+console.log('Enter reporter (e.g. spec, xunit): ');
+
+process.stdin.resume();
+process.stdin.setEncoding('utf8');
+
+process.stdin.on('data', function (type) {
```

Comment:
I'm wondering maybe it will be better to pass reporter as a parameter to the gulp task? E.g.:`gulp report-design-viewer -spec`?